packages/itwin/grouping-mapping-widget/src/components/context/ExtractionStatusDataContext.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { createContext, useContext } from "react";
+
+export interface ExtractionStatusData {
+  iconStatus: "negative" | "positive" | "warning" | undefined;
+  iconMessage: string;
+}
+
+export interface ExtractionMessageData {
+  date: string;
+  category: string;
+  level: string;
+  message: string;
+}
+
+export interface IExtractionStatusDataProps {
+  extractionStatusIcon: ExtractionStatusData;
+  extractionMessageData: ExtractionMessageData[];
+  setExtractionStatusIcon: (extractionStatusIcon: ExtractionStatusData | ((extractionStatusIcon: ExtractionStatusData) => ExtractionStatusData)) => void;
+  setExtractionMessageData: (extractionMessageData: ExtractionMessageData[]) => void;
+}
+
+export const ExtractionStatusDataContext = createContext<IExtractionStatusDataProps>({
+  extractionStatusIcon: {
+    iconStatus: undefined,
+    iconMessage: "",
+  },
+  extractionMessageData: [{
```

Comment:
Wouldn't it be better to have the default `extractionMessageData` set to an empty array? Is there anything that relies on `extractionMessageData` being a non-empty array?

Reply:
I don't think it would affect anything, so I have changed it to an empty array.